[env] add architecture type info to the fastlane env

Kukurijek opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue following this format:
Resolves #999999
-->
Resolves #19532
### Description
<!-- Describe your changes in detail. -->
<!-- Please describe in detail how you tested your changes. -->
New variable `Architecture` for architecture type in the `fastlane env` (macOS only).
For more information please see #19532

### Testing Steps
<!-- Optional: steps, commands, or code used to test your changes. -->
<!-- Providing these will reduce the time needed for testing and review by the fastlane team. -->
run `fastlane env` and check the result - you should be able to find information about your architecture below the swift version (macOS ONLY).

Commit:
add CPU type info to the fastlane env

Commit:
add CPU type info to the fastlane env

joshdholtz commented on fastlane_core/lib/fastlane_core/helper.rb:
```diff
@@ -105,6 +105,12 @@ def self.mac?
+    # @return CPU type
+    def self.cpu_type
+      return "" unless self.mac?
+      return `sysctl -n machdep.cpu.brand_string`.split("\n").first
```
Instead of CPU type, do we actually (or also) want the architecture type? This might be good to see if its running in Rosetta? This can be checked with `uname -p` or `arch`. Thoughts? 😇

Kukurijek replied:
Hi @joshdholtz !
You are completely right. Information about the type of cpu is completely unnecessary if we know what the architecture is. On the other hand, maybe some users may find this information (CPU Info) sensitive, and since we don't need it, I replace it architecture type.

`uname -p` returns a lot of information that is not needed (at least at this point 😄 ), so I changed the command to `arch` which returns only type eg. `i386`

Thanks! 🚀

Kukurijek replied:
Actually `uname -p` returns the same thing as `arch`. I used some other commando before 🤣. Anyway it works and PR is updated.

joshdholtz reviewed (changes requested):
@Kukurijek This is a great idea! Thoughts on either changing or adding if its running in arm or i386 (Rosetta)?

Vannevelj committed:
fixed typo (#19514)

joshdholtz committed:
[action][set_github_release] fix issue where bad status code would print error but fail silently - now actually raises an error (#19516)

Commit:
add architecture type to the fastlane env instead of CPU type

Commit:
add architecture type to the fastlane env instead of CPU type

joshdholtz committed:
[match] nuke now supports filtering by specific certificates (#19584)

* Initial logic to nuke by certificates

* This so good

* Thank you rubocop

* Some improved error handling

* Add interactive check and some more info

joshdholtz committed:
[match][cert] allow developer_id creation if using apple id (#19604)

macblazer committed:
#17667 Added optional `skip_stapling` parameter to `notarize` action. (#19577)

nekrich committed:
[action][get_version_number] - search for MARKETING_VERSION in build settings if target has no INFO_PLIST (#19589)

* get_version_number - search for MARKETING_VERSION in build settings if no plist for target

* Update detail on action about looking at plist and then build settings

Co-authored-by: Josh Holtz <me@joshholtz.com>

taylorfort committed:
[fastlane_core] use -assetFile instead of -f for IPA upload with iTMS (#19596)

* Workaround to validateAssets error for testflight upload

* Undo change to ipa_upload_package_builder, add new argument to transporter for asset file, use asset file if exists and macos

* Fix deliver tests

Co-authored-by: Josh Holtz <me@joshholtz.com>

nekrich committed:
[pem] Add macOS platform support (#19564)

* pem: Add macOS platform support

* A little cleanup

* Fix some logic I broke and rework message

Co-authored-by: Josh Holtz <me@joshholtz.com>

yutailang0119 committed:
[action][set_github_release] Support generate_release_notes parameter (#19558)

yutailang0119 committed:
[action][set_github_release] Fix optional params to guard (#19560)

joshdholtz committed:
Version bump to 2.198.0 (#19611)

joshdholtz committed:
[fastlane_core] fix TransportExecutor to specifically look for ipa, dmg, ipa, and zip and not directory for -assetFile and then fall bask to -f (#19620)

* [fastlane_core] fix TransportExecutor to specifically look for ipa, dmg, ipa, and zip and not directory for -assetFile and then fall bask to -f

* Added a way to force ITMSP if needed and added some more tests

nekrich committed:
[gym] Add xcodebuild_command option (#19614)

christianeiselt committed:
[deliver] Resolves issue uploading iPad Pro 5th gen screenshots (#19616)

* [delivery] iPad Pro 12.9 5th gen. Fix

* [delivery] iPad Pro 12.9 5th gen. Tests

joshdholtz committed:
Version bump to 2.198.1 (#19622)

tmm1 committed:
[action][download_dsyms] use filter argument to /builds to find specific build (#19670)

* [action][download_dsyms] fetch specific version

* [spaceship][connect_api] add iterator that loads pages on demand

* [action][download_dsyms] avoid loading all builds upfront

* [action][download_dsyms] stop processing when uploaded_date has passed; break out of loop because we know results are sorted by -uploadedDate

* [action][download_dsyms] request specific build number too

* Fixed tests

Co-authored-by: Josh Holtz <me@joshholtz.com>

bguidolim committed:
[action][download_dsyms] fix: `download_dsyms` with `wait_for_dsym_processing` is not checking the latest data from Connect API (#19523)

* Fix: `download_dsyms` with `wait_for_dsym_processing` is not checking the latest data from Connect API

* Added tests

* Apply suggestions from code review

Co-authored-by: Satoshi Namai <s.namai.09@gmail.com>

* Fixed tests

Co-authored-by: Josh Holtz <me@joshholtz.com>
Co-authored-by: Satoshi Namai <s.namai.09@gmail.com>

joshdholtz committed:
[tests] fix failing master tests (#19677)

kraai committed:
[deliver][snapshot][fastlane_core] fix typos (#19675)

* Fix misspellings

* Add missing "do"

Commit:
Add examples for simulator descriptions including OS version (#19683)

This reflects the behaviour in https://github.com/fastlane/fastlane/blob/f0b9bbe0c391f034d997ae2ce48bac567c29fa7e/scan/lib/scan/detect_values.rb#L116 and also tests such as https://github.com/fastlane/fastlane/blob/f0b9bbe0c391f034d997ae2ce48bac567c29fa7e/scan/spec/test_command_generator_spec.rb#L671

abcdev committed:
[action] [update_code_signing_settings] add entitlements file path capability (#19678)

justAnotherDev committed:
[spaceship] spaceauth - add new functionality to check if a user's session is still valid (#19658)

* add new flag '--check_session' to spaceship

* add 3 tests for '--check_session' flag

* update documentation to include info about the '--check_session' flag

rogerluan committed:
[action] allow using a `.xcode-version` file with `xcversion` action (#19644)

* Allow using a `.xcode-version` file with xcversion.

* Fix syntax.

* Move .xcode-verison read into default

Co-authored-by: Josh Holtz <me@joshholtz.com>

rogerluan committed:
[docs] modernize markdown syntax of auto-generated README documentation (#19643)

* Modernize markdown syntax of auto-gen docs.

* Fix tests.

joshdholtz committed:
Version bump to 2.199.0 (#19693)

Kukurijek committed:
update license (#19757)

EricAtomic committed:
[scan] Fix: test retries succeeding when they shouldn't (#19749)

* Check for objective c methods when retrying

* Add additional test in failure test

* Fix code style

icecrystal23 committed:
gym: Correctly inherit OTHER_SWIFT_FLAGS when using option 'analyze_build_time' (#19737)

GermanVelibekovHouzz committed:
Fix download-dsym unnecessary builds scan when `after_uploaded_date` provided (#19765)

* [download_dsyms]: break on first older date when after_uploaded_date provided

* [fix-download-dsym] [download_dsyms]: correcting unit tests

Empty2k12 committed:
[deliver] Add Watch Series 7 screen size (#19782)

* Add Series 7 screen size

* fix size

* fix DisplayType::APP_WATCH_SERIES_7

dejunDev committed:
Update get_version_number.rb (#19773)

fix xcode13 plist.info can't contains key ["CFBundleShortVersionString"]

dokimyj committed:
[fastlane_core] use randomly generated filename(s) for -assetFile when uploading binaries to Testflight/App Store (#19716)

* [deliver][pilot] use copied ipa/pkg for -assetFile when uploading binaries with non-alphanumeric filename to Testflight/App Store

* delete unnecessary ifs for assetFile paths

* [test] fl-suggestion

* Fix tests

Co-authored-by: Josh Holtz <me@joshholtz.com>

joeflateau committed:
[supply] add an option to change the release status of a promoted track (#19701)

* add an option to change the release status of a promoted track

* update relevant test

chenshifei committed:
[fastlane_core] fix non-UTF-8 char issues when analysing ipa (#19697)

* Fix non-UTF-8 char issues when analysing ipa

* Fixed case when embedded apps and added tests

* Modified the file member list

* Fix tests on windows

* Skip unzip tests if windows

* Removed extra puts

Co-authored-by: Josh Holtz <me@joshholtz.com>

valerio-castelli committed:
[spaceship] Initial support for updated review submissions flow (#19751)

* feat: add support for review submissions and review submission items

* test: tunes client, review submission and review submission tests

* feat: add new app store version value ready for review

* chore: disable class length warning on app model

* chore: make submission parameters required

joshdholtz committed:
Version bump to 2.200.0 (#19784)

joshdholtz committed:
Remove XCPretty dependency and add xcbeautify as an option (#19629)

* Bring in trainer as an action and standalone tool because trainer logic is needed for scan (#19624)

* Bring in trainer as an action and standalone tool because trainer logic is needed for scan

* Add action category

* This should fix failing tests

* UGH THIS

* Maybe this time

* Replace scan success factor from xcpretty to trainer (#19636)

* Start of thing

* Tests pass again woot woot

* Remove todos

* Fix other failing tests

* omg tests are hard

* Some small fixes from PR review

* Improvements for table output

* Fix trainer tests

* Add native xcode 13 retries (#19695)

* Add native xcode 13 retries

* Clean if statement

* Allow scan to use xcbeautify

* Added tests

* Fix description of xcodebuild_formatter

* Fix tests for ubuntu

* Start of allowing xcbeautify in gym

* [xcbeautify][scan] allow scan to use xcbeautify (#19774)

* Allow scan to use xcbeautify

* Added tests

* Fix description of xcodebuild_formatter

* Fix tests for ubuntu

* This is good now

* Fix test for ubuntu

* Actually fix ubuntu tests

* Bring in trainer as an action and standalone tool because trainer logic is needed for scan (#19624)

* Bring in trainer as an action and standalone tool because trainer logic is needed for scan

* Add action category

* This should fix failing tests

* UGH THIS

* Maybe this time

* Replace scan success factor from xcpretty to trainer (#19636)

* Start of thing

* Tests pass again woot woot

* Remove todos

* Fix other failing tests

* omg tests are hard

* Some small fixes from PR review

* Improvements for table output

* Fix trainer tests

* Add native xcode 13 retries (#19695)

* Add native xcode 13 retries

* Clean if statement

* [xcbeautify][snapshot] snapshot allow xcbeautify (#19786)

* 2.201.0.rc1

* Fix no formatter stuff

* Fix issues where output files were not being created

* Don't run trainer when build_for_testing is set and show output types as warnings with how to get rid of them

* Fix for trainer plist parser

* Give error about removing trainer plugin

* Add new option to scan to ignore retry attempts in output results

* Improve trainer remove retry attempt count

* Added link to doc

* Change version back to 2.200.0

lucgrabowski committed:
[team] Update team.json (#19802)

joshdholtz committed:
Fix Gemfile.lock versions

joshdholtz committed:
Version bump to 2.201.0 (#19822)

joshdholtz committed:
[scan][xcov] set xcresult path in SharedValues and use as default in xcov (#19825)

* [scan][xcov] set xcresult path in SharedValues and use as default in xcov

* Added some safety around if cache exists

* Fixed for tests on Ubuntu

joshdholtz committed:
[trainer] Make new options used by scan public (to fix crash) (#19828)

joshdholtz committed:
[scan] prevent error from raising and prevent xcresult processing when multiple devices with xcpretty (#19829)

* [scan] prevent error from raising and prevent xcresult processing when multiple devices with xcpretty

* Update comment to mention that raw output also works now

joshdholtz committed:
Version bump to 2.201.1 (#19833)

joshdholtz committed:
[scan] fix off by one for scan's number_of_retries with xcode 13 (#19844)

* [scan] fix off by one for scan's number_of_retries with xcode 13

* Make description more clear

joshdholtz committed:
[scan] look for xcresult in derived data instead of forcing temp dir or output_path (#19845)

* [scan] look for xcresult in derived data instead of forcing temp dir or output_path

* Fix crash to build_failure

joshdholtz committed:
[snapshot] fix tests if xcbeautify is installed (#19846)

joshdholtz committed:
Version bump to 2.201.2 (#19847)

joshdholtz committed:
[deliver] migrate to new app store review process (#19838)

* Start of migration deliver to new app store review process

* Fix tests and link

* Updated tests

* Some comments and add in review to in progress submissions

joshdholtz committed:
Version bump to 2.202.0 (#19863)

tejassharma96 committed:
[scan] add parallel_testing option to scan (#19857)

* add parallel_testing option to scan

* Undo Fastlane.swift changes

* Undo changes to fastlane/swift/ScanfileProtocol.swift

Co-authored-by: Josh Holtz <me@joshholtz.com>

tejassharma96 committed:
[scan] Return results when running scan (#19860)

* Return results when running scan

* Updated scan/run_tests return_value and return_type

* Updated so scan will always return results hash

* Cleaned up code

Co-authored-by: Josh Holtz <me@joshholtz.com>

joshdholtz committed:
[scan] fix crash if no derived data (#19871)

joshdholtz committed:
Version bump to 2.203.0 (#19872)

dcacenabes committed:
[trainer][regression] fixes fail_build parameter which stopped working (#19882)

* Fix parsing the parser response

* Add tests

* Flag tests to require Xcode

crazymanish committed:
[scan] Fix issue when scan failed to zip_build_products for `build_for_testing` option (#19875)

Commit:
Fixes bug in string concatenation of verify_build error message (#19873)

kikeenrique committed:
[Fastlane.swift] fixes issues with running on Apple Silicon #18502 (#19555)

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] Update to swift 5

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] Xcode recommedation. Enabling Base Internationalization is recommended for all projects.

* Migrating the “English, deprecated” localization to “English” is recommended for all projects.
   This will ensure localized resources are placed in “en.lproj” directories instead of deprecated “English.lproj” directories.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] Xcode recommendation. Enable recommended warnings.
  Implicit retain of self within blocks
  Overriding deprecated Objective-C methods
  Quoted Include in framework header.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] Xcode recommendation. Enable code signing. It's recommended for macOS executables. This setting will cause executable's code signature to be trusted by your Mac.
   Currently, macOS on M1 is killing the binary at launch. According to console traces, reason is ASP Security policy would not allow process.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] Fix waring Using 'class' keyword to define a class-constrained protocol is deprecated; use 'AnyObject' instead.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] Fix waring 'withUnsafeBytes' is deprecated: use `withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] Fix waring Switch covers known cases, but 'DispatchTimeInterval' may have additional unknown values, possibly added in future versions. Handle unknown values using "@unknown default"

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] Update protocol generator to use AnyObject

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] Update swift package to swift 5

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] Fix thread data race using a thread safe storage

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] On verbose mode, allow output from the runner thread. It allows traces and ease feedback from fastlane swift.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Fastlane.Swift] Replace Build Phase - ShellScript with Build Phase - CopyFiles. This allows to copy the executable signed.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Testing] Add some useful help to debug problems on FastlaneSwiftRunner

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Testing] Fix ensure_actions_config_items_formatting

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Fix style

* Update fastlane/lib/fastlane/swift_lane_manager.rb

Include review comments

Co-authored-by: Roger Oba <rogerluan.oba@gmail.com>

* Update fastlane/lib/fastlane/swift_lane_manager.rb

Include review comments

Co-authored-by: Roger Oba <rogerluan.oba@gmail.com>

* Update Testing.md

Include review comments

Co-authored-by: Roger Oba <rogerluan.oba@gmail.com>

* Include review comments

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Replace DispatchQueue with os_unfair_lock as suggested in review.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Stick to swift 4, as suggested by reviewers.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Keep naming to AtomicDictionary as suggested by reviewers.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Fix CI detected error

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Implement subscript as suggested by reviewers.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Restore values previous to this PR

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Integrate the required steps to only upgrade the Xcode project build phase and flags needed.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Fix CI. Package deployment set to macOS 10.12

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Fix CI warnings.

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Add again support for swift-tools-version 4.0 implementing a OSSpinLockAtomicDictionary for macos versions under 10.12

* Add Atomic Dictionary generic capabilities

* [Fastlane.Swift] Swift fastlane does not run on Apple Silicon #18502

* [Swift] Add Atomic.swift to upgrade_manifest.json

* [Fastlane.swift] Overriding doesn't call specialized extensions, fallback to #available

* Cleanup hacky code

Co-authored-by: Roger Oba <rogerluan.oba@gmail.com>
Co-authored-by: Jorge Revuelta Herrero <minuscorp@gmail.com>
Co-authored-by: Josh Holtz <me@joshholtz.com>

lucgrabowski committed:
[match] Add an option to safe remove certs for nuke (#19617)

* [match][nuke] Add an option to remove certificates from repository without revoking them

* Fix typo in a comment

* [match][nuke] Adjust printed messages for removing certs from repository without revoking them

* [match][nuke] Adjust logs for enterprise accounts

* [match][nuke] Improvements

dweekly committed:
[snapshop] update example  devices included with xcode 13.1 (#19615)

* Update to Devices Included With XCode 13.1

The example devices (6s & iPad Air) are old enough that this demo won't work out of the box anymore with a modern XCode (13.1); suggest updating these to simulators running a current OS (iOS 15.0) and ones included with XCode.

* Try to reset simulators on circlci

* Need to force

* Lets try 3rd generation ipad air

* Maybe this

* I don't know what else to try

* Dude

* I dont think this will work

* I'm not winning tonight

* Fix tests

* Fix bad merge on two tests

Co-authored-by: Josh Holtz <me@joshholtz.com>
Co-authored-by: Lukasz Grabowski <luc.grabowski@gmail.com>

lucgrabowski committed:
[pilot][upload_to_testflight] Send pkg to mac testflight when both ipa and pkg are available (#19801)

* [pilot][upload_to_testflight] Send pkg to mac testflight when both ipa and pkg are available

* [pilot][upload_to_testflight] Add tests for uploading pkg when both ipa and pkg are available

* Fix test for uploading pkg - add platform check in wait_for_build_processing_to_be_complete

* Add tests for wait_for_build_processing_to_be_complete when ipa and pkg files are available

* Improve tests

* Improve tests

fotos committed:
Fix typo in Match::Storage::S3Storage#s3_object_path (#19870)

horvathtamasattila committed:
[pilot] support internal groups for testflight (#19735)

* feat: adding submit_beta_review option

* fix: adjust conditions

* fix: CI fails

* Removed Fastlane.swift changes

* Removed Fastlane.swift changes

Co-authored-by: Josh Holtz <me@joshholtz.com>

MakakWasTaken committed:
[snapshot][fastlane_core] Hotfix/simulator not starting Xcode 13 (#19531)

* Made simulators start when using XCode 13

* Run simulator from uninstall_app function

* Removed whitespace

* Use xcrun instead of Simulator.app

* Revert "Removed whitespace"

This reverts commit 029347bb81579ea8c4326305a48f1321d639540c.

* Revert back to using Simulator.app

* Removed whitespace

* Now using Helper.xcode_at_least

* forgot the question mark

* Require xcode for tests

Co-authored-by: Josh Holtz <me@joshholtz.com>

joshdholtz committed:
Version bump to 2.204.0 (#19886)

joshdholtz committed:
[spaceship][deliver][pilot] temporarily fix finding app by filtering by bundle id locally (#19900)

joshdholtz committed:
Version bump to 2.204.1 (#19901)

joshdholtz committed:
Revert "[spaceship][deliver][pilot] temporarily fix finding app by filtering by bundle id locally (#19900)" (#19906)

This reverts commit f200f0fa05229cef26c0b9e833fa3c5cc25fe176.

joshdholtz committed:
Version bump to 2.204.2 (#19907)

Lutzifer committed:
Update options.rb (#19938)

I'm not sure if this is the right documentation now, but it looks like it was a copy and paste error before.

Hais committed:
[Fastlane.Swift] readPodspec: return map of [String: Any] (#19953)

* readPodspec: return map of String-Any

* Change Ruby action return type to hash

Co-authored-by: Josh Holtz <me@joshholtz.com>

philipparndt committed:
fix compile error on macCatalyst (#19917)

igor-makarov committed:
[pem][spaceship] update development push certificate type ID (#19879)

* update development push certificate type ID

* update test stubs

kikeenrique committed:
[Fastlane.Swift] Swift fastlane upgrader #18933 (#19914)

+ fix upgrades checks to avoid false not up-to-date messages.

igor-makarov committed:
[trainer] Identify skipped tests in `xcresult` and export to Junit format (#19957)

* identify skipped tests and report to junit

* Fully connect scan with skipped tests

Co-authored-by: Josh Holtz <me@joshholtz.com>

joshdholtz committed:
Version bump to 2.204.3 (#19958)

Lutzifer committed:
fix typos in comments and code (#20005)

sebskuse committed:
[trainer] Return Test Plan Configuration in JUnit output (#20016)

* Return Configuration name in Trainer

* Return Configuration name as a property in JUnit

* Extract method to bring under complexity limit

* Mark test as requiring xcode

tkorri committed:
[supply] Prefer all default tracks (#19926) (#20010)

Prefer tracks in production, beta, alpha, internal order when
searching for a release version from Play Store.

VrsajkovIvan33 committed:
[spaceship] Add unresolved_issues to review submission states and update get_in_progress method (#19979)

AliSoftware committed:
[action][git_commit action] use safer way of calling `sh` (#19978)

* git_commit action: Use safer way of calling `sh`

By letting `sh` do all the escaping and argument handling for us

* Fix git_commit_spec

rogerluan committed:
Add env var to skip printing plugins table. (#19963)

att55 committed:
fix typo (#19962)

joshdholtz committed:
[spaceship] replace teams endpoints (#20083)

* [spaceship] Replace apple endpoints used for teams with olympus session

* Replace contentProviderId when selecting a team

* Replaced another instance of contentProviderId

* Tests pass

* A little cleanup

Co-authored-by: Ivan Vrsajkov <ivav@bendingspoons.com>

joshdholtz committed:
Version bump to 2.205.0 (#20084)

nomunomu0504 committed:
[spaceship][tunes] fix 'block in select_team': [!] undefined method `[]' for nil:NilClass (#20088)

* [spaceship] fix undefined method `[]' of temas

* fixed the reference key for access to array.

tahirmt committed:
[trainer] reports failure with skipped tests (#20045)

Trainer would fail if there are any skipped tests in case of retries.

Commit:
Added case-sensitive .DS_Store to gitignore (#20090)

joshdholtz committed:
Version bump to 2.205.1 (#20114)

AliSoftware committed:
match: Avoid mutating app_identifiers (#20125)

casperson committed:
Add support for iPhone 13 device family with updated colors (#20148)

tahirmt committed:
[trainer] Fix missing test failures and improve junit format (#20105)

* Update xcresult parsing logic

* Update tests

* Update junit format to use proper file name and line

Updating the junit format to use the file name and line number according to specs. This allows creating a junit report that can be used to point to the exact file and line in code

* Robocop fixes

joshdholtz committed:
Revert "[trainer] Fix missing test failures and improve junit format (#20105)" (#20159)

This reverts commit 81277cd984d38a9d1a21a12ce2033d1e03516aac.

andersonvom committed:
[spaceship] refresh token and retry on authorization errors (#19502)

* Refresh token and retry on authorization errors

Apple seems to be revoking tokens before their expiration dates. Several
users reporting problems on issue [#19072][1] were likely affected by
it. Sometimes changing the token duration will work around the problem,
as the tokens are likely being revoked later in the future and allowing
the wait to complete.

This can be seen in practice by adding a `@token.refresh!` couple of extra log
messages to [api_client.rb][2] right at the point where we get an
`UnauthorizedAccessError`:

```
INFO  [2021-10-19 18:38:31.74]: Successfully uploaded the new binary to App Store Connect
INFO  [2021-10-19 18:38:31.74]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
INFO  [2021-10-19 18:38:31.74]: Note that if `skip_waiting_for_build_processing` is used but a `changelog` is supplied, this process will wait for the build to appear on AppStoreConnect, update the changelog and then skip the remaining of the processing steps.
DEBUG [2021-10-19 18:38:31.82]: App Platform (ios)
INFO  [2021-10-19 18:38:31.92]: Waiting for processing on... app_id: 1370986669, app_version: 2.23.2, build_version: 262025, platform: IOS
WARN  [2021-10-19 18:38:32.26]: Read more information on why this build isn't showing up yet - https://github.com/fastlane/fastlane/issues/14997
INFO  [2021-10-19 18:38:32.26]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
INFO  [2021-10-19 18:39:02.70]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)

token expired?: false  - expiration: 2021-10-19 18:45:04 -0400
Token has expired, has been revoked, or is invalid! Trying to refresh

INFO  [2021-10-19 18:39:33.26]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
INFO  [2021-10-19 18:40:03.69]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
INFO  [2021-10-19 18:40:34.33]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
INFO  [2021-10-19 18:41:04.76]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
INFO  [2021-10-19 18:41:35.21]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
INFO  [2021-10-19 18:42:05.67]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)

token expired?: false  - expiration: 2021-10-19 18:47:52 -0400
Token has expired, has been revoked, or is invalid! Trying to refresh

INFO  [2021-10-19 18:42:36.34]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
INFO  [2021-10-19 18:43:06.80]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
INFO  [2021-10-19 18:43:37.23]: Waiting for the build to show up in the build list - this may take a few minutes (check your email for processing issues if this continues)
```

The logs above show that, even though the current token is not expired, the
appconnect API responds with a 401 and the process would otherwise fail.

This refreshes the token and adds retries to `UnauthorizedAccessError`
before ultimately raising the error if it still continues to fail.

[1]: https://github.com/fastlane/fastlane/issues/19072
[2]: https://github.com/fastlane/fastlane/blob/62af236780a9eace9d0487d0767f50d1a17a1c6c/spaceship/lib/spaceship/connect_api/api_client.rb#L172-L175

* Added a test

Co-authored-by: Josh Holtz <me@joshholtz.com>

nekrich committed:
[match] Check validity of provisioning profile certificates (#20194)

lucgrabowski committed:
[update_project_provisioning] Fail when profile cannot be verified (#20169)

* [update_project_provisioning] Fail when profile cannot be verified

* update_project_provisioning] Fail when profile cannot be verified - add a hint

* Moved verify logic to new method and added tests

Co-authored-by: Josh Holtz <me@joshholtz.com>

joshdholtz committed:
Version bump to 2.205.2 (#20203)

joshdholtz committed:
[infra] changelog will now show github usernames which makes contributors stand out more in releases (#20214)

liamnichols committed:
[spaceship] update `Spaceship::ConnectAPI::User model` to include `delete!` method (#20251)

* Update Spaceship::ConnectAPI::User model to include delete! method

* Move delete! method above get_visible_apps

lucgrabowski committed:
[spaceship] update buildDeliveries request to use app id in path (#20268)

* [spaceship] Update buildDeliveries request to use app_id in path instead of using it as a part of filter

* Remove nil on required app_id parameter

Co-authored-by: Josh Holtz <me@joshholtz.com>

joshdholtz committed:
[fastlane] allows aliased tool names (build_app, sync_code_signing, etc) to run from CLI (#20287)

* [fastlane] allows aliased tool names (build_app, sync_code_signing, etc) to run from CLI

* Needed to remove a puts

alexanderwe committed:
[action][danger] Add GitHub enterprise flags to Danger action (#20216)

Add tests for  GitHub enterprise flags and fail-if-no-pr flag

javigines committed:
Added timeout param into setup_ci (#20211)

trsneed committed:
Add option to skip google account confirmation (#20223)

nnsnodnb committed:
[spaceship] fix creation App Store Connect API authorization token (#20206)

* Fix creation jwt token for connect_api

* Update spaceship/spec/connect_api/token_spec.rb

t2 committed:
Remove named parameter from `create_capability` example. (#20197)

tahirmt committed:
[scan] fix scan failing to return results when there are test failures (#20237)

* Fix scan failing to return results when there are test failures

When using test_failure! method it causes scan to skip sending the output even if fail_build is set to false

* Also had to check test exit status

Co-authored-by: Josh Holtz <me@joshholtz.com>

polpielladev committed:
[deliver] add `verify_only` option to deliver (#20247)

* Adds `build_verify_command` to itunes transporter executor

* Adds `verify` command to itunes_transporter.rb

* Adds `verify` option and counterpart `DELIVER_VERIFY` env variable

* Adds verify execution to runner.rb and halts after verifying verifying

* Passes provider through to `build_verify_command` methods

* Updates unit tests for `itunes_transporter`

* Updates unit tests for `deliver`’s `runner`

* Adds messaging when `verify` flag is set

* Updates wording of raised error

* Corrects indentation issues with `rubocop`

* Updates option for failing test in `Windows` machines

* Reverts any changes made to `project.pbxproj`

* Rename :verify to :verify_only

* Update error message from ipa to a generic binary

Co-authored-by: Josh Holtz <me@joshholtz.com>

LcTwisk committed:
[spaceship] Add ability to invite a single TestFlight user (#20112)

triplef committed:
[match] Add support for Developer ID certificates from G2 Sub-CA (#20145)

* [match] Add support for Developer ID certificates from G2 Sub-CA

* Return multiple cert types when fetching

Co-authored-by: Josh Holtz <me@joshholtz.com>

joshdholtz committed:
Version bump to 2.206.0 (#20288)

lacostej committed:
[specs] Fix import_from_git running on environment with non master default branch (#20297)

On git >= 2.28.0, force the branch to be master to match the tests expectations.

crazymanish committed:
[Regression][fastlane] Fixed fastlane command issue when tool name not provided (#20295)

* [Regression][fastlane] Fixed regression issue when tool name not provided

* PR review suggestion

joshdholtz committed:
Version bump to 2.206.1 (#20308)

lucgrabowski committed:
[fastlane][tests] add tests for CLIToolsDistributor (#20315)

* [fastlane] Add a test for CLI for map_aliased_tools

* Add TOOL_ALIASES hash with mapping

* Generate cases for all tool aliases

* Run rubocop

lacostej committed:
Test on Ruby 3.1 (#20313)

kamimi01 committed:
fix a grammatical mistake (#20307)

rogerluan committed:
Fixed encoding issue when saving p12 file. (#20317)

SvenTiigi committed:
[Fastlane.Swift] Use ArgumentProcessor port to start fastlane socket_server (#20176)

* Use port of ArgumentProcessor to start socket_server

* Update fastlane/swift/MainProcess.swift

Co-authored-by: Jorge Revuelta <minuscorp@gmail.com>

Co-authored-by: Jorge Revuelta <minuscorp@gmail.com>

joshdholtz committed:
[Fastlane.swift] fix compile issue with argumentProcessor (#20318)

joshdholtz committed:
Version bump to 2.206.2 (#20320)

milch committed:
Allow overriding the architecture to run xcodebuild under (#20323)

lucgrabowski committed:
Update rack due to security alert from dependabot (#20339)

EmDee committed:
Fix typo from uplaod to upload (#20331)

joshdholtz committed:
[Ruby] Ruby 2.6 is now the minimum (dropping Ruby 2.5) (#20413)

* [Ruby] Ruby 2.6 is now the minimum (dropping Ruby 2.5)

* Update CircleCI to use Ruby 2.6 and drop Ruby 2.5 job

* Install python on validate_docs Ruby 2.6 image

* Maybe use fastlanetools docker image

kylehowells committed:
[snapshot] Add missing iPod touch to Generated Snapshot Reports (#20337)

* Add missing iPod touch to Generated Snapshot Reports

* Fix whitespace

iaia committed:
Specify s3 object prefix (#20344)

crazymanish committed:
[action][update_info_plist] improve the plist file-path option validation check (#20356)

* [action][update_info_plist] Improve the plist file-path option validation check

* [action][update_info_plist] PR review suggestion

denrase committed:
Add device Google Pixel 5 (#20389)

crazymanish committed:
[pilot] added the Pilot::TesterExporter class unit tests (#20394)

* [pilot] added the Pilot::TesterExporter class unit tests

* Fix Lint warning.

crazymanish committed:
[fastlane-core][scan] added destination param support inside the `xcodebuild` command (#20399)

* [fastlane-core] Using destination param in `xcodebuild` command to fix the annoying warnings

* Updated unit-tests

* Put logic inside begin-rescue block to avoid shellescape issue

* Fix Linting

* Added unit-tests for destination parameter support for Xcode13+

* Ignore destination warnings for Xcode13+

crazymanish committed:
[match] Added Match::CommandsGenerator's import option unit test (#20396)

crazymanish committed:
[match] Improved Match::CommandsGenerator's decrypt option unit test (#20395)

darbyfrey committed:
[Match] Adding support for GitLab Secure Files as a Match storage mode (#20386)

* Adding support for GitLab Secure Files as a Match storage mode

* Adds warning when both JOB_TOKEN and ACCESS_TOKEN are defined

kalinjul committed:
Show http error message from AppStore (#20397)

Currently, Fastlane swift only shows a ruby stacktrace when a http error message rises, which is useless:

```
FastlaneCore::Interface::FastlaneError:
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/pilot/lib/pilot/build_manager.rb:19:in `upload'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/actions/upload_to_testflight.rb:34:in `run'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/runner.rb:229:in `chdir'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/server/socket_server_action_command_executor.rb:73:in `run'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/server/socket_server_action_command_executor.rb:42:in `execute'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/server/socket_server.rb:181:in `execute_action_command'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/server/socket_server.rb:176:in `process_action_command'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/server/socket_server.rb:123:in `handle_action_command'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/server/socket_server.rb:86:in `parse_and_execute_command'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/server/socket_server.rb:62:in `block in receive_and_process_commands'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/server/socket_server.rb:49:in `loop'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/server/socket_server.rb:49:in `receive_and_process_commands'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/server/socket_server.rb:158:in `listen'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/server/socket_server.rb:38:in `start'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/commands_generator.rb:167:in `block (2 levels) in run'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/commands_generator.rb:354:in `run'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/commands_generator.rb:43:in `start'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/fastlane/lib/fastlane/cli_tools_distributor.rb:123:in `take_off'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/gems/fastlane-2.206.2/bin/fastlane:23:in `<top (required)>'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/bin/fastlane:25:in `load'
/Users/julian/.brew/Cellar/fastlane/2.206.2/libexec/bin/fastlane:25:in `<main>'
[1655712352.5108628]: Done running lane: deliverAppStore 🚀
```

With this change, the important http message "No ipa or pkg file given" is added to the beginning of the error output, too.

DuMaM committed:
[sigh][match][spaceship] add missing inhouse to mac profile types (#20353)

* Add missing inhouse to mac profile types

* Only download mac inhouse certs if apple id auth

* Fix for match when using enterprise and added some nuke tests

Co-authored-by: Josh Holtz <me@joshholtz.com>

lucgrabowski committed:
[spaceship] Allow for removing app from sale by updating territories (#20324)

* [spaceship] Allow for removing app from sale by updating territories with an empty array of ids.

* Remove parenthesis

* [deliver] Fix upload_price_tier to not remove app from sale by updating territories with an empty array of ids.

* [spaceship] Add documentation

* Add allow_removing_from_sale param to decide if empty array allows for removing apps from sale

* Run rubocop

joshdholtz committed:
[match] add imports, fix linting errors, rename env variable (#20418)

snatchev committed:
[deliver] update deliver/runner.rb to handle both ipa and pkg paths. (#20043)

* Update deliver/runner.rb to handle both ipa and pkg paths.

#19659

* Update runner.rb

* reformat

* Update runner.rb

* Updated verify to use same logic

* Fix verify message when no valid platform

Co-authored-by: Josh Holtz <me@joshholtz.com>

PaulTaykalo committed:
[match] add profile_type filtering when checking if profile exists (#20311)

* [match] add profile_type filtering when checking if profile exists

* Moved nuke spec to module spec

Co-authored-by: Josh Holtz <me@joshholtz.com>

joshdholtz committed:
Version bump to 2.207.0 (#20421)

darbyfrey committed:
[match] Fixes set up bugs when using GitLab Secure Files as a Match storage backend (#20452)

* Fixes bug causing errors when uploading files to GitLab Secure Files

* Adding additional handling for file upload errors

* Adding secure files to fastlane match init

* Better error response handling when logging file upload errors

wallisch committed:
[scan] Fix ERB.new() deprecation warnings (#20440)

ZevEisenberg committed:
[snapshot] Use empty string to blank out operator name by default (#20429)

* Use explicit space to blank out carrier name.

* Clarify that you need to set override_status_bar to true in order for override_status_bar_arguments to work.

mbernson committed:
[action][changelog_from_git_commits] Fix description for documentation (#20430)

The documentation says that valid values for the parameter `merge_commit_filtering`
in this action are Ruby symbols. But it actually expects the value to be a string.
This change updates the action's description to make this match.

tejassharma96 committed:
extract the certificate name from provisioning profiles (#20187)

getaaron committed:
Add CircleCI connect instructions to PR template (#20453)

DuMaM committed:
[match] [sigh] Add productsign to import script command (#20474)

DuMaM committed:
[match][sigh] Prefer default keychain during wwdr cert installation (#20448)

* Fix priority of checks in WWDR

* Use latest wwdr

* Remove deprecated cert

* Update tests

* Apply suggestions from code review

Co-authored-by: Aaron Brager <getaaron@gmail.com>

* Rename function to match it's behaviour

* Expect only one cert

* Remove redundant code

* Delete fake testers_file_path

* Update fastlane_core/lib/fastlane_core/cert_checker.rb

* Apply linter fixes

* Make sure password is invalid for decryption

Co-authored-by: Aaron Brager <getaaron@gmail.com>

joshdholtz committed:
[spaceship] fix `fastlane init` and temporarily retrofitting `Spaceship::Tunes::Application.find` (#20480)

* Commiting before i go crazy

* This better hopefully

* Fixed require

* Test lint

* Lint this

* Lint this

* All tests green please

* Huge comment block

* Adding a deprecation message in console

joshdholtz committed:
Version bump to 2.208.0 (#20483)

* Version bump to 2.208.0

* Fix failing test

getaaron committed:
[match] suppress null byte message (#20497)

tremblay committed:
[pilot] app_version and app_build should not be fetched from a local IPA or PKG when distribute_only is set (#20488)

* app_version and app_build should not be fetched from a local IPA or PKG when distribute_only is set

* add a test for the new behavior, remove a duplicate test

DuMaM committed:
Don't create in lazy manner keychain files (#20478)

Co-authored-by: Aaron Brager <getaaron@gmail.com>

aibaars committed:
Fix unescaped \ (#20508)

polmum committed:
Adding no overwrite and local only options to the pod repo push command (#20455)

Co-authored-by: Pablo Muina <pablom@etermax.com>

grey442 committed:
[match][sigh][cert] Added checking hash of installed wwdr certificates (#20507)

* Added checking hash of installed wwdr certificates

Co-authored-by: Aaron Brager <getaaron@gmail.com>

guidev committed:
faceook.design moved to design.facebook.com (#20513)

rogerluan committed:
Fix typo caused by string concatenation. (#20531)

joshdholtz committed:
Version bump to 2.209.0 (#20550)

harrimaatta committed:
Ignoring case sensitive when requesting SMS (#20436)

* Ignoring case sensitive when requesting SMS

* retrigger checks

mofmofmofneko committed:
The word phrasing used has been standardized. (#20543)

e.g. MatchfileTemplate.

Co-authored-by: Yusuke Arakawa <nekolaboratory@users.noreply.github.com>

crazytonyli committed:
[fastlane_core] install all Apple WWDR Intermediate Certificates (#20537)

* Install all Apple WWDRCA certificates instead of just G6

* Use the openssl rubygem to get certficiate subject

* Install 'WWDRCA - G1' certificate

* 'G2' certificate has a different OU

* Use sha256 to identify WWDR certificates

joshdholtz committed:
Version bump to 2.209.1 (#20582)

rogerluan committed:
Fix typo and improve phrasing of documentation. (#20585)

freddi-kit committed:
[deliver][pilot] use altool instead of using iTMSTransporter for Xcode 14 (#20631)

* Use altool instead of using iTMSTransporter for Xcode 14

* fix linter error

* fix linter error

* fix linter error

* use keyword function for transporter_for_selected_team

Co-authored-by: Kohki Miki <giginet.net@gmail.com>

* fix typo and move platform string handling

* remove unecessary return

* create should_use_altool

* fix to use unless to make code one line

Co-authored-by: Kohki Miki <giginet.net@gmail.com>

* make code oneline by using unless

Co-authored-by: Kohki Miki <giginet.net@gmail.com>

* improve error message

* fix to support apple connect api key

* fix jwt check

* fix to set? for boolean function

* fix rubocup issue

* fix rubocup issue

* fix test

* fix to call upload without replacement

* fix typo

* fix checking apikey available or not

* fix to generate p8 file for every calling altool

* PR review fixes (Remove unused file, autogenerated by uni-tests + improved verbose logs)

* Fix itunes_transporter_spec.rb which was failing in Xcode14 https://github.com/fastlane/fastlane/pull/20631/files#r967653482

* Fix helper_spec.rb which was failing in Xcode14 https://github.com/fastlane/fastlane/pull/20631/files#r967653482

* support base64 key_content for using altool with appstore connect apikey

* fix typo

* fix to remove un-necessary begin

Co-authored-by: Olivier Halligon <olivier@halligon.net>

* fix to use unless

Co-authored-by: Olivier Halligon <olivier@halligon.net>

* improve showing error on UI

Co-authored-by: Olivier Halligon <olivier@halligon.net>

* fix to remove warning in altool executer

* improve error log

* add some small fix

* fix code format

* fix to remove unnecessary return

Co-authored-by: Kohki Miki <giginet.net@gmail.com>

* fix to use File.open() |p8|

* fix to use mktmpdir for saving p8 temp

* fix linter issue

* fix to use wb option when making p8 file

* add unit test for altool case

* fix test format

* remove space

* support api json path

* add some comment and fix code style

* add comment about error info

* compose return expression in one line

Co-authored-by: Kohki Miki <giginet.net@gmail.com>

* fix to use checking raise_error

Co-authored-by: Kohki Miki <giginet.net@gmail.com>

* use context for matching rpec

* fix code's readability

* fix lint

* fix lint

* fix lint

* fix ids provider problem

* add comments

* add test for provider ids

* fix api key condition to make code more readable

* add more detailed comment

* fix typo

* improve specifying substring

* fix index

* fix altool providers output to parse json

* fix code format

* fix hash problem if user pass key via environment variable

* remove unnecessary !

* fix null handle

* dummy commit for test

* dummy commit for test

* change the implementation place of provider id parser

* make absract method

* add raise for not implemented functions

Co-authored-by: Kohki Miki <giginet.net@gmail.com>
Co-authored-by: Manish Rathi <i.am.manish.rathi@gmail.com>
Co-authored-by: Olivier Halligon <olivier@halligon.net>

joshdholtz committed:
Version bump to 2.210.0 (#20660)

freddi-kit committed:
[pilot][deliver] Fix how to encode base64 p8 key on deliver and pilot (#20669)

* fix how to encode base64

* fix code format

Co-authored-by: Olivier Halligon <olivier@halligon.net>

* fix rubocop warning

Co-authored-by: Olivier Halligon <olivier@halligon.net>

joshdholtz committed:
Version bump to 2.210.1 (#20682)

cherpake committed:
[deliver] add support to detecting and uploading 6.7" (iPhone 14 Pro Max) screenshots (#20694)

* upload 6.7inch screenshots

* Update deliver/spec/app_screenshot_spec.rb

Co-authored-by: Roger Oba <rogerluan.oba@gmail.com>

* fix

* fix

* Update deliver/lib/deliver/app_screenshot.rb

Co-authored-by: Kohki Miki <giginet.net@gmail.com>

* Update deliver/spec/app_screenshot_spec.rb

Co-authored-by: Kohki Miki <giginet.net@gmail.com>

* Update deliver/lib/deliver/app_screenshot.rb

Co-authored-by: Kohki Miki <giginet.net@gmail.com>

* Update deliver/lib/deliver/app_screenshot.rb

Co-authored-by: Kohki Miki <giginet.net@gmail.com>

* Update deliver/lib/deliver/app_screenshot.rb

Co-authored-by: Kohki Miki <giginet.net@gmail.com>

* revert Gemlock

* Fix indendation.

Co-authored-by: Roger Oba <rogerluan.oba@gmail.com>
Co-authored-by: Kohki Miki <giginet.net@gmail.com>

Commit:
Merge branch 'env-add-cpu-type' of github.com:Kukurijek/fastlane into env-add-cpu-type

iBotPeaches commented:
Going to close this. Something looks odd with history and full of unrelated changes. Sorry we never got this merged in time.

Kukurijek commented:
> Going to close this. Something looks odd with history and full of unrelated changes. Sorry we never got this merged in time.

Thanks for taking the time to go through all the old PRs!

In this case, the branch history got polluted when the old remote branch was merged into a newer local branch, which pulled in a lot of unrelated commits. This is no longer a hot topic, but the change itself could still be useful. It would just need to be recreated on a fresh branch.

Have fun working on fastlane 🚀

iBotPeaches commented:
Yep it was probably this commit in isolation is all we need - https://github.com/fastlane/fastlane/pull/19533/changes/bd51ac0eaf2a9fe7fee1ce50cbab18696a415de0

Saying that though, with Apple taking Intel out of equation shortly. I'm less worried about knowing architecture when I'd only really want to know a few years ago when it was heavily mixed with intel vs apple silicon.
